cli/flox-rust-sdk/src/models/manifest.rs

```diff
@@ -1,24 +1,233 @@
+    ///
+    /// Additionally, this method prefixes each table with documentation on its usage, and
+    /// and inserts commented configuration examples for tables left empty.
+    pub fn new_documented(
+        systems: &[&System],
+        customization: &InitCustomization,
+        use_catalog: bool,
+    ) -> RawManifest {
+        let mut manifest = DocumentMut::new();
+
+        // `version` number
+        if use_catalog {
+            manifest.insert(MANIFEST_VERSION_KEY, toml_edit::value(1));
+        }
+
+        // `[install]` table
+        let packages_vec = vec![];
+        let packages = customization.packages.as_ref().unwrap_or(&packages_vec);
+
+        let mut install_table = if packages.is_empty() {
+            // Add comment with example packages
+            let mut table = Table::new();
+
+            table.decor_mut().set_suffix(indoc! {r#"
+
+                # hello.pkg-path = "hello"
+                # nodejs = { version = "^18.4.2", pkg-path = "nodejs_18" }"#
+            });
+
+            table
+        } else {
+            Table::from_iter(packages.iter().map(|pkg| (&pkg.id, InlineTable::from(pkg))))
+        };
+
+        install_table.decor_mut().set_prefix(indoc! {r#"
+
+            # List packages you wish to install in your environment inside
+            # the `[install]` section.
+        "#});
+
+        manifest.insert(MANIFEST_INSTALL_KEY, Item::Table(install_table));
+
+        // `[vars]` table
+        let mut vars_table = Table::new();
+
+        vars_table.decor_mut().set_prefix(indoc! {r#"
+
+            # Set environment variables in the `[vars]` section. These variables may not
+            # reference once another, and are added to the environment without first
+            # expanding them. They are available for use in the `[profile]` and `[hook]`
+            # scripts.
+        "#});
+
+        // [sic]: vars not customized using InitCustomization yet
+        vars_table.decor_mut().set_suffix(indoc! {r#"
+
+            # message = "Howdy""#});
+
+        manifest.insert(MANIFEST_VARS_KEY, Item::Table(vars_table));
+
+        // `[hook]` table
+        let mut hook_table = Table::new();
+
+        hook_table.decor_mut().set_prefix(indoc! {r#"
+
+            # The `hook.on-activate` script is run by the *bash* shell immediately upon
+            # activating an environment, and will not be invoked if Flox detects that the
+            # environment has previously been activated. Variables set by the script will
+            # be inherited by `[profile]` scripts defined below. Note that any stdout
+            # generated by the script will be redirected to stderr.
+        "#});
+
+        if let Some(ref hook_on_activate_script) = customization.hook_on_activate {
+            let on_activate_content = indent::indent_all_by(4, hook_on_activate_script);
+
+            hook_table.insert("on-activate", toml_edit::value(on_activate_content));
```

Comment:
suggestion blocking: the Python activate hook doesn't have a trialing newline, so I'm seeing:
```
    # End autogenerated by Flox"""
```
The `"""` should be on the next line. We probably need to change the init hooks, not code here though